yellow-paper/docs/public-vm/state.md

```diff
@@ -0,0 +1,107 @@
+# State
+
+This section describes the types of state maintained by the AVM.
+
+## Machine State
+
+**Machine state** is transformed on an instruction-per-instruction basis. Each execution context has its own machine state.
+
+### _MachineState_
+
+| Field                 | Type            | Description |
+| ---                   | ---             | ---         |
+| `l1GasLeft`           | `field`         | Tracks the amount of L1 gas remaining at any point during execution. |
+| `l2GasLeft`           | `field`         | Tracks the amount of L2 gas remaining at any point during execution. |
+| `daGasLeft`           | `field`         | Tracks the amount of DA gas remaining at any point during execution. |
+| `pc`                  | `field`         | Index into the contract's bytecode indicating which instruction to execute. Initialized\* to 0. |
+| `internalCallStack`   | `Vector<field>` | A stack of program counters pushed to and popped from by `INTERNALCALL` and `INTERNALRETURN` instructions. Initialized\* as empty. |
+| `memory`              | `[field; 2^32]` | A $2^{32}$ entry memory space accessible by user code (bytecode instructions). All 2^32 entries are initialized\* to 0. See ["Memory Model"](./memory-model) for a complete description of AVM memory. |
+
+## World State
+
+### AVM's access to Aztec State
+Aztec's global state is comprised of a few databases most of which have merkle tree implementations. For a comprehensive guide to Aztec's global state, see ["State"](../state). For a guide to state as it pertains to contract classes and instances see ["Contract Deployment"](../contract-deployment).
```

Comment:
Really one data base with a few different trees. Also all of them are trees, rather than most of them

Reply:
> For a guide to state as it pertains to contract classes and instances

Is this the sort of tone we are going for, this feels a little docs like

Reply:
Good call, this doesn't match YP tone